evo_researcher/benchmark/agents.py

```diff
@@ -114,7 +119,49 @@ def predict(self, market_question: str, researched: str) -> t.Optional[Predictio
             return None
 
 
+class RephrasingOlasAgent(OlasAgent):
+    def __init__(
+        self,
+        model: str,
+        temperature: float,
+        agent_name: str = "reph-olas",
+        max_workers: t.Optional[int] = None,
+        embedding_model: EmbeddingModel = EmbeddingModel.spacy,
+        use_negated_question: bool = True,
+        use_universal_question: bool = True,
+    ):
+        super().__init__(
+            model=model,
+            temperature=temperature,
+            embedding_model=embedding_model,
+            agent_name=agent_name,
+            max_workers=max_workers,
+        )
+        self.use_negated_question = use_negated_question
+        self.use_universal_question = use_universal_question
+
+    def research(self, market_question: str) -> t.Optional[str]:
+        questions = rephrase_question(question=market_question)
+
+        report_original = super().research(market_question=questions.original_question)
```

Comment:
> otherwise this research report will have up to 3x the tokens

It doesn't have to be fair in the amount of tokens, does it? Maybe this agent has more information because it was able to find it thanks to the negated/open-ended questions, but that was the goal.

Reply:
Yeah I guess so, as it seems like it's set to an arbitrary number (for cost reasons). Just wanted to make the point that changes in outcome from adding extra context might be from just having more tokens as well as having rephrased questions.